<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Shamit Soneji" />

<meta name="date" content="2018-06-07" />

<title>A guide to using cellexalvrR</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">A guide to using cellexalvrR</h1>
<h4 class="author"><em>Shamit Soneji</em></h4>
<h4 class="date"><em>2018-06-07</em></h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This document describes how to use <code>cellexalvrR</code>, an R package that accompanies <a href="wwww.cellexalvr.med.lu.se">CellexalVR</a> which is a virtual reality environment to analyse single-cell RNAseq data. <code>cellexalvrR</code> has two functions:</p>
<ol style="list-style-type: decimal">
<li>To aid the formatting and export of data that can be imported by CellexalVR.</li>
<li>To perform bakend calculations during a CellexalVR session.</li>
</ol>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The easiest way to install <code>cellexalvrR</code> is directly from github using the <code>devtools</code> package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(devtools)
<span class="kw">install_github</span>(<span class="st">&quot;sonejilab/cellexalvrR&quot;</span>)</code></pre></div>
<p><strong>If you are installing this on your VR station then be sure to install <code>cellexalvrR</code> system-wide, so on your Windows 10 machine right-click the R icon, and then “Run as administator” before issuing the commands above.</strong></p>
</div>
<div id="quick-start" class="section level2">
<h2>Quick start</h2>
<p>The data from <a href="http://www.bloodjournal.org/content/128/8/e20.long?sso-checked=true">Nestorowa <em>et al</em></a> can be downloaded from <a href="https://www.cellexalvr.med.lu.se/download#">here</a>. Unpack them and set your working directory to where they are.</p>
<p>First, load the library:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(cellexalvrR)</code></pre></div>
<p>Then load the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;log2data.RData&quot;</span>)
<span class="kw">load</span>(<span class="st">&quot;facs.RData&quot;</span>)
<span class="kw">load</span>(<span class="st">&quot;cell.ids.RData&quot;</span>)
<span class="kw">load</span>(<span class="st">&quot;diff.proj.RData&quot;</span>)
<span class="kw">load</span>(<span class="st">&quot;ddr.proj.RData&quot;</span>)
<span class="kw">load</span>(<span class="st">&quot;tsne.proj.RData&quot;</span>)</code></pre></div>
<p><code>log2data</code> is a matrix of highly variable genes in log2 form. The first 10 columns and 10 rows look as so:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">log2data[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]
<span class="co">#&gt;                 HSPC_001 HSPC_002  HSPC_003 HSPC_004 HSPC_006 HSPC_008</span>
<span class="co">#&gt; Clec1b         0.0000000 0.000000 0.0000000 0.000000 1.039889 1.172368</span>
<span class="co">#&gt; Kdm3a          0.0000000 0.000000 0.0000000 0.000000 0.000000 8.313856</span>
<span class="co">#&gt; Coro2b         9.4755769 0.000000 0.0000000 0.000000 1.637906 1.172368</span>
<span class="co">#&gt; 8430408G22Rik  0.0000000 0.000000 0.0000000 0.000000 0.000000 0.000000</span>
<span class="co">#&gt; Clec9a         0.0000000 0.000000 0.0000000 0.000000 0.000000 0.000000</span>
<span class="co">#&gt; Phf6           7.7333697 7.271829 9.4130405 2.677008 5.832172 7.856124</span>
<span class="co">#&gt; Usp14          1.4788998 3.378561 8.6774851 1.659063 7.391105 3.091340</span>
<span class="co">#&gt; Tmem167b       0.0000000 9.093340 0.8302297 0.000000 1.637906 0.000000</span>
<span class="co">#&gt; Kbtbd7        10.0456005 2.861341 1.3538522 1.659063 2.385231 7.902221</span>
<span class="co">#&gt; Rag2           0.5329056 2.047347 0.8302297 0.000000 6.381515 0.000000</span>
<span class="co">#&gt;               HSPC_009 HSPC_011 HSPC_012 HSPC_014</span>
<span class="co">#&gt; Clec1b        0.000000 0.000000 1.486056 0.000000</span>
<span class="co">#&gt; Kdm3a         7.042431 2.732774 2.202400 2.209467</span>
<span class="co">#&gt; Coro2b        0.000000 1.090491 0.000000 0.000000</span>
<span class="co">#&gt; 8430408G22Rik 0.000000 0.000000 0.000000 0.000000</span>
<span class="co">#&gt; Clec9a        0.000000 0.000000 0.000000 0.000000</span>
<span class="co">#&gt; Phf6          3.298544 9.103079 6.695850 1.491864</span>
<span class="co">#&gt; Usp14         8.689676 7.566973 2.202400 8.129944</span>
<span class="co">#&gt; Tmem167b      0.000000 0.000000 1.486056 1.491864</span>
<span class="co">#&gt; Kbtbd7        0.000000 0.000000 0.000000 0.000000</span>
<span class="co">#&gt; Rag2          0.000000 1.704399 7.365545 1.491864</span></code></pre></div>
<p>Note the cell IDs are in the column names, and the gene names are the row names.</p>
<p><code>facs</code> is a matrix of cell surface marker intensities captured during index sorting:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(facs)
<span class="co">#&gt;               CD34      CD16     c.Kit      EPCR      Flk2     CD150</span>
<span class="co">#&gt; HSPC_001 0.6037796 0.4322826 1.3509535 1.5235524 0.3618872 0.3036440</span>
<span class="co">#&gt; HSPC_002 0.3652949 0.5923082 1.1668190 1.5575771 0.3985330 0.1802330</span>
<span class="co">#&gt; HSPC_003 1.1688350 0.3103070 1.2716602 0.1796609 0.2225457 0.2708689</span>
<span class="co">#&gt; HSPC_004 0.7234537 0.5098244 1.2890254 0.1047645 0.7382823 0.1746147</span>
<span class="co">#&gt; HSPC_006 0.7806266 0.2244971 0.8613574 0.1909399 0.7124364 0.4767371</span>
<span class="co">#&gt; HSPC_008 0.8303475 0.5984071 1.2414646 1.4069329 0.4826786 0.2322792</span>
<span class="co">#&gt;               CD48       Lin      Sca1</span>
<span class="co">#&gt; HSPC_001 0.4359241 0.2989843 1.3179900</span>
<span class="co">#&gt; HSPC_002 0.3413969 0.4922320 1.5137737</span>
<span class="co">#&gt; HSPC_003 1.4005640 0.4194964 0.6391425</span>
<span class="co">#&gt; HSPC_004 1.0903352 0.5070497 0.5363241</span>
<span class="co">#&gt; HSPC_006 1.1907472 0.3140937 0.5699483</span>
<span class="co">#&gt; HSPC_008 0.3781136 0.2573233 1.2604954</span></code></pre></div>
<p>Cell IDs are in the row names, and the name of the surface protein in the columns.</p>
<p><code>cell.ids</code> is a 1/0 matrix that assigns metadata to the each cell. In this case it shows the cell type each cell was sorted as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(cell.ids)
<span class="co">#&gt;          LTHSC_broad LMPP_broad MPP_broad CMP_broad MEP_broad GMP_broad</span>
<span class="co">#&gt; HSPC_001           0          0         1         0         0         0</span>
<span class="co">#&gt; HSPC_002           1          0         0         0         0         0</span>
<span class="co">#&gt; HSPC_003           0          0         1         0         0         0</span>
<span class="co">#&gt; HSPC_004           0          1         0         0         0         0</span>
<span class="co">#&gt; HSPC_006           0          1         0         0         0         0</span>
<span class="co">#&gt; HSPC_008           0          1         0         0         0         0</span>
<span class="co">#&gt;          MPP1_broad MPP2_broad MPP3_broad STHSC_broad LTHSC LMPP MPP CMP</span>
<span class="co">#&gt; HSPC_001          0          0          0           1     0    0   0   0</span>
<span class="co">#&gt; HSPC_002          0          0          0           0     1    0   0   0</span>
<span class="co">#&gt; HSPC_003          0          0          1           0     0    0   0   0</span>
<span class="co">#&gt; HSPC_004          0          0          0           0     0    1   0   0</span>
<span class="co">#&gt; HSPC_006          0          0          0           0     0    1   0   0</span>
<span class="co">#&gt; HSPC_008          0          0          0           0     0    1   0   0</span>
<span class="co">#&gt;          MEP GMP MPP1 MPP2 MPP3 STHSC ESLAM HSC1 Projected</span>
<span class="co">#&gt; HSPC_001   0   0    0    0    0     0     0    0         0</span>
<span class="co">#&gt; HSPC_002   0   0    0    0    0     0     0    0         0</span>
<span class="co">#&gt; HSPC_003   0   0    0    0    0     0     0    0         0</span>
<span class="co">#&gt; HSPC_004   0   0    0    0    0     0     0    0         0</span>
<span class="co">#&gt; HSPC_006   0   0    0    0    0     0     0    0         0</span>
<span class="co">#&gt; HSPC_008   0   0    0    0    0     0     0    0         0</span></code></pre></div>
<p>Cell IDs are in the row names, and the name of the meta data in the columns. A 1 is given if the cell belongs to a metadata class, 0 otherwise.</p>
<p><code>diff.proj</code>,<code>tsne.proj</code>,and <code>diff.proj</code> are the results from three different dimension reduction methods applied to <code>log2data</code>, specifically DDRTree, tSNE and diffusion map respectively. Each is a three column matrix of x/y/z coordinates. For example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(diff.proj)
<span class="co">#&gt;              [,1]         [,2]         [,3]</span>
<span class="co">#&gt; [1,] -0.014760110 -0.024119177 -0.007998293</span>
<span class="co">#&gt; [2,] -0.010255308 -0.019970919 -0.015754327</span>
<span class="co">#&gt; [3,] -0.003649320  0.027297654  0.014202316</span>
<span class="co">#&gt; [4,] -0.012653205 -0.002990825  0.025813077</span>
<span class="co">#&gt; [5,] -0.005876981  0.021537658  0.035740656</span>
<span class="co">#&gt; [6,] -0.015699148 -0.018777106  0.007916241</span></code></pre></div>
<p>The first step is to put all the dimension reduction outputs into a single list:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">proj.list &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">diffusion=</span>diff.proj,<span class="dt">DDRtree=</span>ddr.proj,<span class="dt">tSNE=</span>tsne.proj)</code></pre></div>
<p>The next is to make a <code>cellexalvr</code> object by calling <code>MakeCellexaVRObj</code> and passing the required objects to it:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cellvr &lt;-<span class="st"> </span><span class="kw">MakeCellexaVRObj</span>(log2data,<span class="dt">mds.list=</span>proj.list,<span class="dt">specie=</span><span class="st">&quot;mouse&quot;</span>,<span class="dt">cell.meta=</span>cell.ids,<span class="dt">facs.data=</span>facs)</code></pre></div>
<p>In the same step we also set the specie as mouse which ensures the correct transcription factor IDs are using during network construction if implemented in CellexalVR.</p>
<p>Calling the object name will detail it’s contents:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cellvr
<span class="co">#&gt; An object of class cellexalvr </span>
<span class="co">#&gt; with 4709 genes and 1654  cells. </span>
<span class="co">#&gt; Annotation datasets (0,0): ''   </span>
<span class="co">#&gt; Sample annotation (1654,23): 'LTHSC_broad', 'LMPP_broad', 'MPP_broad', 'CMP_broad', 'MEP_broad', 'GMP_broad', 'MPP1_broad', 'MPP2_broad', 'MPP3_broad', 'STHSC_broad', 'LTHSC', 'LMPP', 'MPP', 'CMP', 'MEP', 'GMP', 'MPP1', 'MPP2', 'MPP3', 'STHSC', 'ESLAM', 'HSC1', 'Projected'   </span>
<span class="co">#&gt; There are 0 user groups stored :</span>
<span class="co">#&gt; and 3 mds object(s)</span>
<span class="co">#&gt; Specie is set to mouse</span></code></pre></div>
<p>The last step is to call <code>export2cellexalvr</code> which will write the neccessary files from the <code>cellvr</code> object to a specified directory:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">export2cellexalvr</span>(cellvr,<span class="st">&quot;CellexalOut/&quot;</span>)</code></pre></div>
<p>All the files needed by CellexalVR are created at this point. The entire “CellexalOut/” folder should then be moved/copied to the “Data” folder in your CellexalVR setup. See the <a href="https://www.cellexalvr.med.lu.se/manual">manual</a> for more details including a schematic of the folder structure.</p>
</div>
<div id="making-a-cellexalvr-object-from-scratch" class="section level2">
<h2>Making a <code>cellexalvr</code> object from scratch</h2>
<p>While <code>MakeCellexaVRObj</code> is the most convenient way to make the object, sometimes you want to make one manually. This is done calling <code>new</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell.vr.scr &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;cellexalvr&quot;</span>,<span class="dt">data=</span>log2data,<span class="dt">mds=</span><span class="kw">list</span>(<span class="dt">tsne=</span>tsne.proj))
cell.vr.scr
<span class="co">#&gt; An object of class cellexalvr </span>
<span class="co">#&gt; with 4709 genes and 1654  cells. </span>
<span class="co">#&gt; Annotation datasets (0,0): ''   </span>
<span class="co">#&gt; Sample annotation (0,0): ''   </span>
<span class="co">#&gt; There are 0 user groups stored :</span>
<span class="co">#&gt; and 1 mds object(s)</span>
<span class="co">#&gt; Specie is set to</span></code></pre></div>
<p>We can add another set of dimension reduction coordinates using the <code>addMDS2cellexalvr</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell.vr.scr &lt;-<span class="st"> </span><span class="kw">addMDS2cellexalvr</span>(cell.vr.scr,ddr.proj,<span class="st">&quot;DDRTree&quot;</span>)
cell.vr.scr
<span class="co">#&gt; An object of class cellexalvr </span>
<span class="co">#&gt; with 4709 genes and 1654  cells. </span>
<span class="co">#&gt; Annotation datasets (0,0): ''   </span>
<span class="co">#&gt; Sample annotation (0,0): ''   </span>
<span class="co">#&gt; There are 0 user groups stored :</span>
<span class="co">#&gt; and 2 mds object(s)</span>
<span class="co">#&gt; Specie is set to</span></code></pre></div>
<p>To add metadata for each cell use <code>addCellMeta2cellexalvr</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell.vr.scr &lt;-<span class="st"> </span><span class="kw">addCellMeta2cellexalvr</span>(cell.vr.scr,cell.ids)
cell.vr.scr
<span class="co">#&gt; An object of class cellexalvr </span>
<span class="co">#&gt; with 4709 genes and 1654  cells. </span>
<span class="co">#&gt; Annotation datasets (0,0): ''   </span>
<span class="co">#&gt; Sample annotation (1654,23): 'LTHSC_broad', 'LMPP_broad', 'MPP_broad', 'CMP_broad', 'MEP_broad', 'GMP_broad', 'MPP1_broad', 'MPP2_broad', 'MPP3_broad', 'STHSC_broad', 'LTHSC', 'LMPP', 'MPP', 'CMP', 'MEP', 'GMP', 'MPP1', 'MPP2', 'MPP3', 'STHSC', 'ESLAM', 'HSC1', 'Projected'   </span>
<span class="co">#&gt; There are 0 user groups stored :</span>
<span class="co">#&gt; and 2 mds object(s)</span>
<span class="co">#&gt; Specie is set to</span></code></pre></div>
<p>To add FACS for each cell use <code>addFACS2cellexalvr</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell.vr.scr &lt;-<span class="st"> </span><span class="kw">addFACS2cellexalvr</span>(cell.vr.scr,facs)</code></pre></div>
<p>Setting the specie is done using the <code>set.specie</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell.vr.scr &lt;-<span class="st"> </span><span class="kw">set.specie</span>(cell.vr.scr,<span class="st">&quot;mouse&quot;</span>)
cell.vr.scr
<span class="co">#&gt; An object of class cellexalvr </span>
<span class="co">#&gt; with 4709 genes and 1654  cells. </span>
<span class="co">#&gt; Annotation datasets (0,0): ''   </span>
<span class="co">#&gt; Sample annotation (1654,23): 'LTHSC_broad', 'LMPP_broad', 'MPP_broad', 'CMP_broad', 'MEP_broad', 'GMP_broad', 'MPP1_broad', 'MPP2_broad', 'MPP3_broad', 'STHSC_broad', 'LTHSC', 'LMPP', 'MPP', 'CMP', 'MEP', 'GMP', 'MPP1', 'MPP2', 'MPP3', 'STHSC', 'ESLAM', 'HSC1', 'Projected'   </span>
<span class="co">#&gt; There are 0 user groups stored :</span>
<span class="co">#&gt; and 2 mds object(s)</span>
<span class="co">#&gt; Specie is set to mouse</span></code></pre></div>
</div>
<div id="making-cell-metadata-from-a-data-frame" class="section level2">
<h2>Making cell metadata from a data frame</h2>
<p>CellexalVR requires metadata in the form of a 1/0 matrix, but many packages store it as a data frame. cellexalvrR has function to convert a data frame into a 1/0 matrix. First, lets make a data frame:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meta.df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">CellType=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">&quot;Type1&quot;</span>,<span class="st">&quot;Type2&quot;</span>,<span class="st">&quot;Type3&quot;</span>),<span class="dv">10</span>,<span class="dt">replace=</span>T),
                      <span class="dt">Phase=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">&quot;G1&quot;</span>,<span class="st">&quot;G2M&quot;</span>,<span class="st">&quot;S&quot;</span>),<span class="dv">10</span>,<span class="dt">replace=</span>T),
                      <span class="dt">Treatment=</span><span class="kw">sample</span>(<span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Dox&quot;</span>),<span class="dv">10</span>,<span class="dt">replace=</span>T))
<span class="kw">head</span>(meta.df)
<span class="co">#&gt;   CellType Phase Treatment</span>
<span class="co">#&gt; 1    Type2   G2M        WT</span>
<span class="co">#&gt; 2    Type1     S        WT</span>
<span class="co">#&gt; 3    Type1   G2M       Dox</span>
<span class="co">#&gt; 4    Type3   G2M       Dox</span>
<span class="co">#&gt; 5    Type1   G2M       Dox</span>
<span class="co">#&gt; 6    Type1     S       Dox</span></code></pre></div>
<p>We can now make a correctly formatted cell metadata matrix by applying the function <code>make.cell.meta.from.df</code> using only the fields we need, in this case the “CellType” and “Treatment” columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
required.cell.metad &lt;-<span class="st"> </span><span class="kw">make.cell.meta.from.df</span>(meta.df,<span class="kw">c</span>(<span class="st">&quot;CellType&quot;</span>,<span class="st">&quot;Treatment&quot;</span>))
<span class="kw">head</span>(required.cell.metad)
<span class="co">#&gt;      CellType.Type2 CellType.Type1 CellType.Type3 Treatment.WT</span>
<span class="co">#&gt; [1,]              1              0              0            1</span>
<span class="co">#&gt; [2,]              0              1              0            1</span>
<span class="co">#&gt; [3,]              0              1              0            0</span>
<span class="co">#&gt; [4,]              0              0              1            0</span>
<span class="co">#&gt; [5,]              0              1              0            0</span>
<span class="co">#&gt; [6,]              0              1              0            0</span>
<span class="co">#&gt;      Treatment.Dox</span>
<span class="co">#&gt; [1,]             0</span>
<span class="co">#&gt; [2,]             0</span>
<span class="co">#&gt; [3,]             1</span>
<span class="co">#&gt; [4,]             1</span>
<span class="co">#&gt; [5,]             1</span>
<span class="co">#&gt; [6,]             1</span></code></pre></div>
<p>It can be seen the field name is placed in the column heading preceeding a “.”, and this is used by CellexalVR to form catagories on the menu system, so “CellType” and “Treatment” will be on separate tabs. This metadata matrix can now be added to a <code>cellexalvr</code> object as decribed above using the <code>addCellMeta2cellexalvr</code> function.</p>
</div>
<div id="converting-a-seurat-object-to-a-cellexalvr-object" class="section level2">
<h2>Converting a Seurat object to a cellexalvr object</h2>
<p>For those using <a href="https://satijalab.org/seurat/">Seurat</a>, cellexalvrR offers some basic functionality to convert a Seurat object to a cellexalvr object. It takes the data from the <code>@data</code> slot, and cell identities from the <code>@ident</code> slot, and will also take the cell-cycle phase assignments if this has been done. If PCA has been performed the first three components are taken as MDS coordinates, and the same if tSNE has been implemented using the <code>dim.embed=3</code> option when calling <code>RunTSNE</code> from Seurat.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cellvrObj &lt;-<span class="st"> </span><span class="kw">seurat2cellexalvr</span>(SeuratObj)</code></pre></div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
